doc: child_process: clarify `subprocess.pid` can be `undefined` when `ENOENT` (#37014)
nodejs/node: doc/api/child_process.md

> * {integer}

Reviewer: ```suggestion
* {integer|undefined}
```

Returns the process identifier (PID) of the child process. If the child process

> fails to spawn due to errors like command or `cwd` not found (`ENOENT`), then

Reviewer: Sorry, another suggested revision here.

```suggestion
fails to spawn due to errors, then
```

> failed to spawn due to errors like command not found (`ENOENT`), the value will
> be `undefined`, and `error` should emit on next tick.

Reviewer: ```suggestion
fails to spawn due to errors like command not found (`ENOENT`), then the value is
`undefined` and `error` should emit on next tick.
```

> the value is `undefined` and `error` should emit on next tick.

Reviewer: ```suggestion
the value is `undefined` and `error` is emitted.
```
